### PR TITLE
Adding ViT, VGG,. and two MLP models for TinyImageNet 

### PR DESCRIPTION
**Pull request by augustinasjucas**
### Added some models
I have added the support of the following models:
- VGG13 with no norm
- VGG 13 with Continual Norm
- Vision transformer
*^ These 3 models have been tuned to the extremest, both achieving around 55% test-set accuracy on the full TinyImageNet dataset.*
- Simple MLP, achieving around 13% accuracy on the test set.
- MLPMixer model, achieving 36% on the test set.

Also could be extremely easily extended to other VGGs (I also have hyperarameter data for them), and also for the use of BatchNorm. A large number of runs used for selecting hyperparameters can be found [on my wandb](https://wandb.ai/augustinasj/Clean%20TinyImageNet%20Model%20Comparisons/panel/3fegqdonu?nw=nwuseraugustinasjucas).

### Adding future models
Made the addition of new models very easy, for every new model, one just has to provide TrainingParameters() (that stores hyperarameters) and the MyNNModule object, which has a `get_named_layers_list()` function. And also, all submodules of the model should implement reset_parameters() method (very easy to do!).

### Augmentations
Updated the training scripts to support CutMix and MixUp augmentations. In particular, CutMix proved to have a MASSIVE advantage for training ViTs.  Added a list of other augmentations used for training the models. Some of them taken from other papers (links at the top of the files), some done by myself.

### Performant TinyImageNet - pickle files
Added the support of TinyImageNet. In particular, I converted the dataset into two pickle files - one for training one for validation. That is primarily because the I/O on Euler is **extremely** slow and loading the data from the images directly takes up to 30 minutes (provided you pre-load the data into RAM before startingtraining, cause if you don't then it's even worse).  So instead, I loaded the data into an object and then exported that object into a pickle file. Then this file is just loaded at the just before training. Since the pkl file is a single file, with consecutive bytes, it is read from the IO very fast. This way, we save 20-30 minutes for every run. 

As a result, you need to clone the repository using git-lfs. If you clone just by git, the pkl files will not actually download properly. If at some point we will want to change the pkl files, I can take care of that.

### CODE REVIEW

**Review by jankulik — APPROVED** [review]
LGTM, I didn't find any obvious bugs
